src/ORM/Behavior/Translate/ShadowTableStrategy.php

```diff
@@ -111,11 +111,11 @@ protected function setupAssociations()
      * and adding a formatter to copy the values into the main table records.
      *
      * @param \Cake\Event\EventInterface $event The beforeFind event that was fired.
-     * @param \Cake\Datasource\QueryInterface $query Query.
+     * @param \Cake\ORM\Query $query Query.
      * @param \ArrayObject $options The options for the query.
      * @return void
      */
-    public function beforeFind(EventInterface $event, QueryInterface $query, ArrayObject $options): void
+    public function beforeFind(EventInterface $event, Query $query, ArrayObject $options): void
```

Comment:
Now that's sad if we must go back to concrete objects.

Reply:
@dereuromark we can add interfaces in another PR and update the definitions later.